shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1800,12 +1867,14 @@ - (void)hoverEvent:(UIPanGestureRecognizer*)recognizer API_AVAILABLE(ios(13.4))
   CGFloat scale = [UIScreen mainScreen].scale;
   _mouseState.location = {location.x * scale, location.y * scale};
 
-  flutter::PointerData pointer_data = [self generatePointerDataForMouse];
+  flutter::PointerData pointer_data =
+      [self generateDataForHoverPointerWithDeviceKind:flutter::PointerData::DeviceKind::kMouse];
 
   pointer_data.signal_kind = flutter::PointerData::SignalKind::kNone;
   packet->SetPointerData(/*index=*/0, pointer_data);
 
   [_engine.get() dispatchPointerDataPacket:std::move(packet)];
+  _mouseState.hover_pointer_is_added = pointer_data.change != flutter::PointerData::Change::kRemove;
```

Comment:
I would rather not change it. With new revision it's not being modified at all.

Reply:
I mean does the revision change the semantics of this variable from a "mouse" state to a "mouse+trackpad" state?

Reply:
I don't know why this has its own struct at all. I guess it comes from the mac implementation, where there are a lot of contained variables. And there it is called MouseState, where trackpad has existed on the Mac since that struct's creation. If you insist it can be changed. I would rather keep it the same name. Other names like PointerState and CursorState are confusing since it refers to memory pointer or text cursor.

Reply:
In this case can you explicitly comment out in `MouseState` that it's for both mouse and trackpad.

Reply:
For sure, i will update the comment.